chemprop/models/model.py

```diff
@@ -33,7 +34,16 @@ def __init__(self, args: TrainArgs):
         if self.classification or self.multiclass:
             self.no_training_normalization = args.loss_function in ['cross_entropy', 'binary_cross_entropy']
 
-        self.output_size = args.num_tasks
+        self.is_atom_bond_targets = args.is_atom_bond_targets
+
+        if self.is_atom_bond_targets:
+            self.atom_targets, self.bond_targets = args.atom_targets, args.bond_targets
+            self.atom_constraints, self.bond_constraints = args.atom_constraints, args.bond_constraints
+            self.device = args.device
+            self.output_size = 1
```

Comment:
The `self.output_size` is a relative size multiplier for atom/bond targeting purposes and the base output size is set by the number of task names. This works, but now it means something different for the normal molecule targeting purposes. It would be better if the functions for these were made more parallel.

I think there are a few ways that you could make them parallel. The way that I would recommend would be:

- Make `self.output_size` relative for both of them. Set it initially as 1 for both cases. And maybe change the name to `self.relative_output_size` or something.
- Have the `output_size` input to DenseLayers be `args.num_tasks * self.relative_output_size`
- Change the inputs to MultiReadout to take numbers of tasks instead of the list of task names. That way we're not passing strings down further than we need to when we just need the number of how many, `len(args.atom_targets)` or similar.
- In the constraints loops, loop over the number of tasks rather than the task names.